### DISCUSSION TITLE
❓ What should I do if I want to save weights every 5 cycles? Thanks

### DISCUSSION THREAD

**Discussion by shz1314** [open-edge-platform/anomalib#1615] [Q&A] [answered]
### What is the motivation for this task?

What should I do if I want to save weights every 5 cycles? Thanks

### Describe the solution you'd like

What should I do if I want to save weights every 5 cycles? Thanks

### Additional context

What should I do if I want to save weights every 5 cycles? Thanks

**Comment by blaz-r** [comment, marked as answer]
You will need to modify the following code:
https://github.com/openvinotoolkit/anomalib/blob/1f50c952fc65a165884b2b94e178821dcebfbbef/src/anomalib/utils/callbacks/__init__.py#L63-L69
and add `every_n_epochs` or `every_n_train_steps` or whichever suits you. You might need to remove monitor, but I am not sure. Check [ModelCheckpoint page](https://pytorch-lightning.readthedocs.io/en/1.8.6/api/pytorch_lightning.callbacks.ModelCheckpoint.html) for more details on checkpointing.
